Adds a Custom User Model

daaray opened the pull request:
Work on #2 

Lays the framework to start with a Custom User.
- Sets Email as the identifier
- ~~allows username to be blank~~

~~Once this is reviewed, we will need to see how best to customize https://github.com/torchbox/wagtail/blob/b8b079a28c1792b309ae88343cc9a71f6c0f3c10/wagtail/wagtailusers/views/users.py and its associated forms and templates if we want to further customize the User (remove username altogether, for instance).~~
- Overriding `wagtailusers` forms, views, and urls to allow for custom auth user with email as username

daaray committed:
Adds a Custom User Model

daaray committed:
update org to clone from

Commit:
adding demo site temporarily

Commit:
updating to latest release of wagtail

Commit:
latest merging stylesheets with wagtail demo site

kmtracey commented on accounts/models.py:
```diff
@@ -0,0 +1,70 @@
+    Email and password are required. Other fields are optional.
+    """
+    email = models.EmailField(_('email address'), max_length=254, unique=True)
+    username = models.CharField(_('username'), max_length=254, blank=True)
```
Why do we not just have a field named "username" with definition of email field? Why have both of these fields?

kmtracey committed:
Set execute bit on manage.py

Commit:
latest workon wagtail models, templates

Commit:
Merging changes from diverged branch. Merge branch 'develop' of github.com:rapidpro/rapidpro-community-portal into develop

kmtracey committed:
Merge branch 'develop' into 2-custom-user

(Resolved) Conflicts:
    rapidpro_community_portal/settings/base.py

daaray committed:
Updates to allow for better integration of Custom User Model with Wagtails half-baked support of custom user models

kmtracey committed:
Adjust requiremnts.

Update many to latest, also remove fabric and its deps as those
don't work on Python3.

kmtracey commented:
I merged this branch into develop (which was created subsequent to this pull request).
